Modules/_asynciomodule.c

```diff
@@ -1601,11 +1601,25 @@ static void
 FutureIter_dealloc(futureiterobject *it)
 {
     PyTypeObject *tp = Py_TYPE(it);
-    asyncio_state *state = get_asyncio_state_by_def((PyObject *)it);
+
+    // FutureIter is a heap type so any subclass must also be a heap type.
```

Comment:
This comment is misleading, `FutureIter` is not a basetype so cannot have subclasses.
cc @savannahostrowski